code/modules/mob/living/simple_animal/abnormality/he/funeral.dm

```diff
@@ -0,0 +1,258 @@
+	desc = "An towering abnormality possessing a white butterfly for a head and a coffin on its back."
+	icon = 'ModularTegustation/Teguicons/64x96.dmi' //HOW DO I TURN A PNG INTO THE DMI SPRITES AAAAAAAAAAAAAAA
+	icon_state = "funeral"
+	icon_living = "funeral"
+	icon_dead = "funeral_dead"
+	del_on_death = FALSE
+	maxHealth = 1350 //I am a menace to society.
+	health = 1350
+
+	ranged = TRUE
+	minimum_distance = 0
+
+	move_to_delay = 4
+	damage_coeff = list(BRUTE = 1, RED_DAMAGE = 0.5, WHITE_DAMAGE = 1.5, BLACK_DAMAGE = 1.0, PALE_DAMAGE = 2)
+	armortype = WHITE_DAMAGE
+	stat_attack = HARD_CRIT
+
+
+
+	faction = list("hostile")
+	can_breach = TRUE
+	can_buckle = FALSE
+	vision_range = 14
+	aggro_vision_range = 20
+	threat_level = HE_LEVEL
+	start_qliphoth = 2
+	work_chances = list(
+						ABNORMALITY_WORK_INSTINCT = list(50, 45, 40, 0, 0),
+						ABNORMALITY_WORK_INSIGHT = 50,
+						ABNORMALITY_WORK_ATTACHMENT = 0,
+						ABNORMALITY_WORK_REPRESSION = list(0, 0, 60, 60, 60),
+						)
+	work_damage_amount = 12
+	work_damage_type = WHITE_DAMAGE
+	deathmessage = "floats into the air, descending into its own coffin."
+
+	ego_list = list(
+		/datum/ego_datum/weapon/solemnvow,
+		/datum/ego_datum/weapon/solemnlament,
+		/datum/ego_datum/armor/solemnlament
+		)
+	gift_type =  /datum/ego_gifts/solemnlament
+	var/gun_cooldown
+	var/gun_cooldown_time = 4 SECONDS
+	var/gun_damage = 60
+	var/swarm_cooldown
+	var/swarm_cooldown_time = 30 SECONDS
+	var/swarm_damage = 12
+	var/can_act = TRUE
+/datum/action/innate/abnormality_attack/SpiritGun
+	name = "Bring to Rest"
+	icon_icon = 'icons/obj/wizard.dmi'
+	button_icon_state = "scroll"
+	chosen_message = "<span class='colossus'>You will now fire butterflies.</span>"
+	chosen_attack_num = 1
+
+/datum/action/innate/abnormality_attack/ButterflySwarm
+	name = "Empty Casket"
+	icon_icon = 'icons/obj/wizard.dmi'
+	button_icon_state = "scroll"
+	chosen_message = "<span class='colossus'>You will now unleash a swarm of butterflies.</span>"
+	chosen_attack_num = 2
+/mob/living/simple_animal/hostile/abnormality/funeral/AttackingTarget(atom/attacked_target)
+	return OpenFire()
+
+/mob/living/simple_animal/hostile/abnormality/funeral/OpenFire()
+	if(!can_act)
+		return
+	if(client)
+		switch(chosen_attack)
+			if(1)
+				SpiritGun(target)
+			if(2)
+				ButterflySwarm(target)
+		return
+	if(swarm_cooldown <= world.time && prob(75))
+		ButterflySwarm(target)
+	else
+		SpiritGun(target)
+	return
+
+
+/mob/living/simple_animal/hostile/abnormality/funeral/proc/SpiritGun(atom/target)
+	if(!istype(target, /mob/living)||gun_cooldown > world.time)
+		return
+	var/mob/living/cooler_target = target
+	can_act = FALSE
+	visible_message("<span class='danger'>[src] levels one of its arms at [cooler_target]!</span>")
+	gun_cooldown = world.time + gun_cooldown_time
+	SLEEP_CHECK_DEATH(1 SECONDS)
+	playsound(get_turf(src), 'sound/abnormalities/funeral/spiritgun.ogg', 75, 1, 3)
+	if(cooler_target in oview(src, 12))
```

Comment:
Not funny... Why is it auto-aim without any effects?

Reply:
Because it's an auto-aim without any projectiles in LC?
I'll increase the warning time, but i'm not a spriter so i can't do shit for the attack telegraph aside from screaming at the target.